biz.aQute.bndlib/src/aQute/bnd/osgi/Analyzer.java

```diff
@@ -2528,10 +2528,20 @@ public List<Jar> getClasspath() {
 						warning("Cannot find entry on -classpath: %s", s);
 				}
 		}
-		return classpath;
+		return Collections.unmodifiableList(classpath);
 	}
 
 	public void addClasspath(Jar jar) {
+		addClasspath(jar, -1);
+	}
+
+	public void addClasspath(Jar jar, int release) {
```

Comment:
I think, if one want to make it work, that way, doing it at the time of **adding** classpath won't work well in the long term, but it could work doing it when one is **getting* the classpath!

So `List<Jar> getClasspath(int release)` or even `Jar getClasspath(int release) ` (as one can actually even flatten *all* jars into a single linear jar view if we assume manifests do not matter here). That way one don't destroy the original added classpath jars and can later on analyze the classpath for different release levels.